community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/TransactionBoundPlanContext.scala

```diff
@@ -54,23 +56,26 @@ class TransactionBoundPlanContext(tc: TransactionalContextWrapper, logger: Inter
   def indexExistsForLabel(labelName: String): Boolean = {
     val labelId = tc.statement.readOperations().labelGetForName(labelName)
 
-    val indexDescriptors = tc.statement.readOperations().indexesGetForLabel(labelId).asScala
-    val onlineIndexDescriptors = indexDescriptors.flatMap(getOnlineIndex)
+    val onlineIndexDescriptors = tc.statement.readOperations().indexesGetForLabel(labelId).asScala
+                                  .filter(_.`type`() == KernelIndexDescriptor.Type.GENERAL)
+                                  .flatMap(getOnlineIndex)
 
     onlineIndexDescriptors.nonEmpty
   }
 
   def uniqueIndexesGetForLabel(labelId: Int): Iterator[IndexDescriptor] = {
     // here we do not need to use getOnlineIndex method because uniqueness constraint creation is synchronous
-    tc.statement.readOperations().uniqueIndexesGetForLabel(labelId).asScala.map(kernelToCypher)
+    tc.statement.readOperations().indexesGetForLabel(labelId).asScala
+      .filter(_.`type`() == KernelIndexDescriptor.Type.UNIQUE)
+      .map(kernelToCypher)
   }
 
   def uniqueIndexGet(labelName: String, propertyKeys: Seq[String]): Option[IndexDescriptor] = evalOrNone {
```

Comment:
I didn't filter for `indexGet` or `uniqueIndexGet` because there can be only one index, and whether it is types as unique or not should not matter to Cypher. I think the plan is to remove the unique variants completely from the context and lets the index get versions return all indexes - just like this PR does to the Kernel interfaces.